unlock-app/src/hooks/useCheckoutCommunication.ts

```diff
@@ -19,39 +20,61 @@ export enum CheckoutEvents {
 
 type Payload = UserInfo | TransactionInfo
 
+type Emitter = (kind: CheckoutEvents, payload?: Payload) => void
+
 interface BufferedEvent {
   kind: CheckoutEvents
   payload?: Payload
 }
 
+export class ProviderAdapter {
+  private emit: Emitter
+
+  constructor(emit: Emitter) {
+    this.emit = emit
+  }
+
+  foo() {
+    this.emit(CheckoutEvents.closeModal)
+  }
+}
+
 // This is just a convenience hook that wraps the `emit` function
 // provided by the parent around some communication helpers. If any
 // events are called before the handshake completes, they go into a
 // buffer. After that, once the handle to the parent is available, all
 // the buffered events are emitted and future events are emitted
 // directly.
 export const useCheckoutCommunication = () => {
+  let parent: Postmate.ChildAPI | undefined
   const [buffer, setBuffer] = useState([] as BufferedEvent[])
-  const [config, setConfig] = useState<PaywallConfig | undefined>(undefined)
-  const parent = usePostmateParent({
-    setConfig: (config: PaywallConfig) => {
-      setConfig(config)
-    },
-  })
 
-  const pushOrEmit = (kind: CheckoutEvents, payload?: Payload) => {
+  const pushOrEmit: Emitter = (kind, payload) => {
     if (!parent) {
       setBuffer([...buffer, { kind, payload }])
     } else {
       parent.emit(kind, payload)
     }
   }
 
+  const [config, setConfig] = useState<PaywallConfig | undefined>(undefined)
+  const [providerAdapter, setProviderAdapter] = useState<
+    ProviderAdapter | undefined
+  >(undefined)
+  parent = usePostmateParent({
+    setConfig: (config: PaywallConfig) => {
+      setProviderAdapter(new ProviderAdapter(pushOrEmit))
```

Comment:
when we do this for real, it will only be if the config specifies we're using a provider on the other end